### PR TITLE
Only show ToolTip when pointer still over control.

### PR DESCRIPTION
**Pull request by grokys**
Make sure the pointer is still over the control when the timer fires to show a ToolTip. Previously the ToolTip was showing even if the pointer had been moved away.

This can be seen on `master` by opening DevTools and quickly passing the pointer over a property name. The tooltip will display after the delay even though the pointer is no longer over the property name.

### CODE REVIEW

File: src/Avalonia.Controls/ToolTip.cs [hunk 1]
```diff
@@ -106,24 +106,28 @@ private static void ShowToolTip(Control control)
             if (control != null && control.IsVisible && control.GetVisualRoot() != null)
             {
                 var cp = (control.GetVisualRoot() as IInputRoot)?.MouseDevice?.GetPosition(control);
-                var position = control.PointToScreen(cp ?? new Point(0, 0)) + new Vector(0, 22);
 
-                if (s_popup == null)
+                if (cp.HasValue && control.IsVisible && new Rect(control.Bounds.Size).Contains(cp.Value))
```
Comment (kekekeks) [inline src/Avalonia.Controls/ToolTip.cs]: Why are you checking cursor rect instead of `IsPointerOver`?

Comment (kekekeks) [inline src/Avalonia.Controls/ToolTip.cs]: BTW, static fields are evil. Could we move it somewhere?

Comment (grokys) [inline src/Avalonia.Controls/ToolTip.cs]: It's best to check the rect as `IsPointerOver` can be affected by mouse capture. That was the cause of a bug here: https://github.com/AvaloniaUI/Avalonia/pull/1011.

Comment (grokys) [inline src/Avalonia.Controls/ToolTip.cs]: And yeah we need to remove the static I guess, but i'm not 100% sure what we should do. Create a new `PopupRoot` for each time? One per toplevel? It should be a separate PR whatever.

Comment (kekekeks) [inline src/Avalonia.Controls/ToolTip.cs]: If mouse is captured by another control, we shouldn't show the tooltip. So checking for IsPointerOver is correct behavior, I think

Comment (grokys) [inline src/Avalonia.Controls/ToolTip.cs]: The issue is whether we should show the tooltip if the mouse is captured by the control _with_ the tooltip and the mouse is moved away from the control. Not sure what the correct behavior here is tbh, i'd need to check with other frameworks.
